src/hooks/useStripeTerminal.tsx

```diff
@@ -149,29 +153,40 @@ export function useStripeTerminal(props?: Props) {
   const _isInitialized = useCallback(() => isInitialized, [isInitialized]);
 
   const {
-    onUpdateDiscoveredReaders,
-    onFinishDiscoveringReaders,
+    onDidDisconnect,
     onDidFinishInstallingUpdate,
     onDidReportAvailableUpdate,
+    onDidReportLowBatteryWarning,
+    onDidReportReaderEvent,
     onDidReportReaderSoftwareUpdateProgress,
-    onDidReportUnexpectedReaderDisconnect,
-    onDidStartInstallingUpdate,
-    onDidRequestReaderInput,
     onDidRequestReaderDisplayMessage,
-    onDidChangePaymentStatus,
+    onDidRequestReaderInput,
+    onDidStartInstallingUpdate,
+    onDidUpdateBatteryLevel,
+  } = props?.readerCallbacks || {};
+
+  const {
     onDidChangeConnectionStatus,
+    onDidChangePaymentStatus,
+    onDidReportUnexpectedReaderDisconnect,
+  } = props?.terminalCallbacks || {};
+
+  const {
+    onDidFailReaderReconnect,
     onDidStartReaderReconnect,
     onDidSucceedReaderReconnect,
-    onDidFailReaderReconnect,
+  } = props?.reconnectionCallbacks || {};
+
+  const {
     onDidChangeOfflineStatus,
     onDidForwardPaymentIntent,
     onDidForwardingFailure,
-    onDidDisconnect,
-    onDidUpdateBatteryLevel,
-    onDidReportLowBatteryWarning,
-    onDidReportReaderEvent,
-    onDidAcceptTermsOfService,
-  } = props || {};
```

Comment:
hm I was thinking we would keep this part still as `props || {}` and just change the type names. This way it's less likely to break most integration unless they are doing type comparisons.